pyresample/utils/cf.py

```diff
@@ -0,0 +1,497 @@
+                uniqs[adef] = i
+
+        # return only the indices
+        return uniqs.values()
+
+    adefs = []
+    infos = []
+
+    # go through all the variables
+    for v in nc_handle.variables.keys():
+
+        # skip variables that are less than 2D: they cannot
+        #   possibly sustain an AreaDefinition
+        if nc_handle[v].ndim < 2:
+            continue
+
+        try:
+            # try and load an AreaDefinition from this variable
+            adef, info = _load_cf_area_oneVariable(nc_handle, v)
+            # store
+            adefs.append(adef)
+            infos.append(info)
+            # break the loop, we have all we need
+            break
+        except ValueError:
+            # this is not a problem: variable v simply doesn't define an AreaDefinition
+            continue
+
+    # go through the loaded AreaDefinitions and find the unique ones.
+    indices = _indices_unique_AreaDefs(adefs)
+    uniq_adefs = [adefs[ui] for ui in indices]
+    uniq_infos = [infos[ui] for ui in indices]
+
+    return uniq_adefs, uniq_infos
+
+
+def load_cf_area(nc_file, variable=None, y=None, x=None, with_cf_info=False):
+    """Load an AreaDefinition object from a netCDF/CF file
+
+    Parameters
+    ----------
+    nc_file : string or object
+        path to a netCDF/CF file, or opened xarray.Dataset object
+    variable : string, optional
+        name of the variable to load the AreaDefinition from
+        If variable is None the file will be searched for valid CF
+        area definitions
+    y : string, optional
+        name of the variable to use as 'y' axis of the CF area definition
+        If y is None an appropriate 'y' axis will be deduced from the CF file
+    x : string, optional
+        name of the variable to use as 'x' axis of the CF area definition
+        If x is None an appropriate 'x' axis will be deduced from the CF file
+    with_cf_info : bool, optional
+        also return a cf_info dict which holds useful information from the
+        CF file
+
+    Returns
+    -------
+    area_def : geometry.AreaDefinition object (default)
+    are_def, cf_info : geometry.AreaDefinition object, dict (if with_cf_info == True)
+
+    """
+
+    # basic check on the default values of the parameters.
+    if (x is not None and y is None) or (x is None and y is not None):
+        raise ValueError("You must specify either both or none of x= and y=")
+
+    # the nc_file can be either the path to a netCDF/CF file, or directly an opened xarray.Dataset()
+    if isinstance(nc_file, xr.Dataset):
+        nc_handle = nc_file
+    else:
+        #   if the path to a file, open the Dataset access to it
+        try:
+            nc_handle = xr.open_dataset(nc_file)
+        except FileNotFoundError:
+            raise FileNotFoundError('File not found: {}'.format(nc_file))
+        except OSError:
+            raise ValueError('File is not a netCDF file {}'.format(nc_file))
```

Comment:
Why not let xarray handle these errors ?

Reply:
This is still hanging from when netCDF4 was used to open the dataset. I will check.

Reply:
Let xarray handle the exceptions when opening netcdf file: ae54b54c809cd27c0243b6cca8037a4da6c2438f